lib/cn/csched_sp3.c

```diff
@@ -1104,7 +1105,22 @@ sp3_process_workitem(struct sp3 *sp, struct cn_compaction_work *w)
         }
     }
 
-    sp3_dirty_node_locked(sp, tn);
+    if (split) {
+        for (int i = 0; i < 2; i++) {
+            struct cn_tree_node *node = w->cw_split.nodev[i];
+
+            if (node) {
+                struct sp3_node *spn = tn2spn(node);
+
+                if (!spn->spn_initialized)
+                    sp3_node_init(sp, spn);
+
+                sp3_dirty_node_locked(sp, node);
+            }
+        }
+    } else {
+        sp3_dirty_node_locked(sp, tn);
+    }
```

Comment:
Funny, I had gotten rid of spn_initialized in my patch cuz no one was using it!  Guess I need to revert that..

Reply:
Sorry.. Hope I'm not creating too many merge conflicts for you..

Reply:
Not at all, I think it should merge without much fuss :-)